interface/forms/newpatient/report.php

```diff
@@ -15,25 +18,34 @@
 use OpenEMR\Common\Acl\AclMain;
 use OpenEMR\Services\AppointmentService;
 use OpenEMR\Services\UserService;
+use OpenEMR\Common\Twig\TwigContainer;
 
 function newpatient_report($pid, $encounter, $cols, $id)
 {
     $res = sqlStatement("select e.*, f.name as facility_name from form_encounter as e join facility as f on f.id = e.facility_id where e.pid=? and e.id=?", array($pid,$id));
-    print "<table><tr><td>\n";
+    $twig = new TwigContainer(__DIR__, $GLOBALS['kernel']);
+    $t = $twig->getTwig();
+    $encounters = [];
+    $userService = new UserService();
     while ($result = sqlFetchArray($res)) {
-        $userService = new UserService();
-        $provider = $userService->getUser($result["provider_id"]);
-        $referringProvider = $userService->getUser($result["referring_provider_id"]);
+        $hasAccess = (empty($result['sensitivity']) || AclMain::aclCheckCore('sensitivities', $result['sensitivity']));
+        $rawProvider = $userService->getUser($result["provider_id"]);
+        $rawRefProvider = $userService->getUser($result["referring_provider_id"]);
         $calendar_category = (new AppointmentService())->getOneCalendarCategory($result['pc_catid']);
-        print "<span class=bold>" . xlt('Facility') . ": </span><span class=text>" . text($result["facility_name"]) . "</span><br />\n";
-        if (empty($result['sensitivity']) || AclMain::aclCheckCore('sensitivities', $result['sensitivity'])) {
-            print "<span class=bold>" . xlt('Category') . ": </span><span class=text>" . text($calendar_category[0]['pc_catname']) . "</span><br />\n";
-            print "<span class=bold>" . xlt('Reason') . ": </span><span class=text>" . nl2br(text($result["reason"])) . "</span><br />\n";
-            print "<span>" . xlt('Provider') . ": </span><span class=text>" . text($provider['lname'] . ", " . $provider['fname']) . "</span><br />\n";
-            print "<span>" . xlt('Referring Provider') . ": </span><span class=text>" . text(($referringProvider['lname'] ?? '') . ", " . ($referringProvider['fname'] ?? '')) . "</span><br />\n";
-            print "<span>" . xlt('POS Code') . ": </span><span class=text>" . text(sprintf('%02d', trim($result['pos_code'] ?? ''))) . "</span><br />\n";
-        }
-    }
+        $reason = (!$hasAccess) ? false : nl2br(text($result['reason']));
+        $provider = (!$hasAccess) ? false : text($rawProvider['fname'] . " " . $rawProvider['lname']);
+        $referringProvider = (!$hasAccess) ? false : $rawRefProvider;
+        $posCode = (!$hasAccess) ? false : text(sprintf('%02d', trim($result['pos_code'] ?? false)));
+        $facility_name = (!$hasAccess) ? false : $result['facility_name'];
```

Comment:
recommend instead getting in habit of doing the escaping as far downstream as possible, which in this case would be in the twig template (provides more flexibility of which escaping function will be used (for example, if a future dev decides to place the provider name in a html attribute in the twig template, then much easier to do it there than need to work out how to do it here, which would then essentially require 2 different variables with different escaping) and makes it much easier to audit code for security)

Reply:
to this end, would encapsulate the nl2br function in a twig call